pensieve/statistics.py

```diff
@@ -101,30 +122,47 @@ def transform(self, df: DataFrame, metric: str) -> "StatisticResultCollection":
             individual_summary_quantiles=summary_quantiles,
         )
 
-        ma_individual_result = ma_result["individual"]
-
-        # floating point arithmetic was a mistake
-        lower_index, upper_index = None, None
-        for branch, branch_result in ma_individual_result.items():
-            for i in branch_result.index:
-                try:
-                    f = float(i)
-                except ValueError:
-                    continue
-                if abs(f - summary_quantiles[0]) < 0.0001:
-                    lower_index = i
-                if abs(f - summary_quantiles[1]) < 0.0001:
-                    upper_index = i
+        for branch, branch_result in ma_result["individual"].items():
+            lower, upper = _extract_ci(branch_result, critical_point)
             result = StatisticResult(
                 metric=metric,
-                statistic=self.name(),
+                statistic="mean",
                 parameter=None,
                 label=branch,
                 ci_width=self.confidence_interval,
                 point=branch_result["mean"],
-                lower=branch_result[lower_index] if lower_index else None,
-                upper=branch_result[upper_index] if upper_index else None,
+                lower=lower,
+                upper=upper,
             )
             stats_results.data.append(result)
 
+        for branch, branch_result in ma_result["comparative"].items():
+            lower_abs, upper_abs = _extract_ci(branch_result["abs_uplift"], critical_point)
+            stats_results.data.append(
+                StatisticResult(
+                    metric=metric,
+                    statistic="mean",
+                    parameter=None,
+                    label=f"{branch} - control",
+                    ci_width=self.confidence_interval,
+                    point=branch_result["abs_uplift"]["exp"],
+                    lower=lower_abs,
+                    upper=upper_abs,
+                )
+            )
+
+            lower_rel, upper_rel = _extract_ci(branch_result["rel_uplift"], critical_point)
+            stats_results.data.append(
+                StatisticResult(
+                    metric=metric,
+                    statistic="mean",
```

Comment:
I'm wondering how it'll be possible (e.g. for dashboarding tools) to distinguish the rows representing the relative uplift from the ones representing the absolute uplift in BigQuery. Not sure if that should be part of the statistic name?

Reply:
My suggestion is that `contrast` would contain e.g. `absolute_difference` or `relative_uplift`.

`branch` and `comparison_to_control` (which is NULL if there's no comparison) are probably reasonable names.

Reply:
Sounds good. So should there be another column `contrast` in the proposed schema?

Reply:
I convinced myself to rename them over the course of that comment, sorry -- the current PR has "branch" and "comparison_to_control" (which I was formerly calling "contrast") columns instead of "label".

Reply:
names are hard

<img src="https://media.giphy.com/media/P5wPrhzZDdeJW/giphy.gif">